### PR TITLE
timepicker

### PR DESCRIPTION
**Pull request by pkozlowski-opensource**
This is the first version of a timepicker. What remains to be done in this PR is:
* [x] - look & feel
* [x] - test improvements

After this initial version gets merged we can open issues for:
* [ ] - corner cases (invalid model etc.)
* [ ] - more features (disabled, time-range, ...)
* [ ] - timepicker as drop-down
* [ ] - duration mode

### CODE REVIEW

**Comment by pkozlowski-opensource** [comment]
I believe that this one is ready for the review now, PTAL.

**Comment by icfantv** [comment]
I'll pull down and look at styling when I get to work.

File: src/timepicker/timepicker.ts [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+ */
+@Component({
+  selector: 'ngb-timepicker',
+  styles: [`
+    .chevron::before {
+      border-style: solid;
+      border-width: 0.29em 0.29em 0 0;
+      content: '';
+      display: inline-block;
+      height: 0.70em;
```
Comment (icfantv) [inline src/timepicker/timepicker.ts]: `height` should be `0.69em`

File: src/timepicker/timepicker.ts [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+      transform: rotate(-45deg);
+      vertical-align: middle;
+      width: 0.71em;
+    }
+    
+    .chevron.bottom:before {
+      top: -.3em;
+      transform: rotate(135deg);
+    }
+  `],
```
Comment (icfantv) [inline src/timepicker/timepicker.ts]: need to add CSS:
```
    .btn-link {
      outline: 0;
    }
```
otherwise we get the button halo affect and in IMO looks weird because it's not really a button.

**Comment by icfantv** [comment]
Just the two CSS changes and LGTM.

**Comment by pkozlowski-opensource** [comment]
@icfantv fixed CSS per your suggestion, thnx. @Foxandxss you wanted to have another look at it, right?

File: src/timepicker/timepicker.spec.ts [hunk 3]
```diff
@@ -0,0 +1,369 @@
...
+
+function getButtons(nativeEl: HTMLElement) {
+  return nativeEl.querySelectorAll('button.btn-link');
+}
+
+function getMeridianButton(nativeEl: HTMLElement) {
+  return nativeEl.querySelector('button.btn-primary-outline');
+}
+
+function expectToDisplayTime(el: HTMLElement, time: string) {
```
Comment (Foxandxss) [inline src/timepicker/timepicker.spec.ts]: This is better to be done as a matcher.

**Comment by Foxandxss** [comment]
Sounds good to me. Left a minor comment.

File: src/timepicker/timepicker.spec.ts (hunk 3 above)
Comment (pkozlowski-opensource) [inline src/timepicker/timepicker.spec.ts]: Yes, we should move to Jasmine matchers in place of helper functions for most of our tests. I will open a separate issue for this.
